app/Events/NewPostEvent.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace App\Events;
+
+use App\Entities\Post;
+use App\Entities\Thread;
+use App\Entities\User;
+use App\Models\NotificationSettingModel;
+use App\Models\PostModel;
+
+class NewPostEvent
+{
+    /**
+     * Number of notifications sent.
+     */
+    private int $count = 0;
+
+    public function __construct(protected Thread $thread, protected Post $post)
+    {
+    }
+
+    /**
+     * Get number of notifications sent.
+     */
+    public function getCount(): int
+    {
+        return $this->count;
+    }
+
+    /**
+     * Process event.
+     */
+    public function process(): void
+    {
+        if ($this->thread->author_id !== $this->post->author_id) {
+            // prepare notification settings for thread author
+            $this->handleThreadNotifications();
+        }
+
+        // prepare notification settings for post authors
+        $this->handlePostNotifications();
+    }
+
+    /**
+     * Send email notification.
+     *
+     * @todo Make it run in the background (queueNotification).
+     */
+    private function sendNotification(User $recipient, Thread $thread, Post $post): bool
+    {
+        helper('text');
+
+        return service('email', false)
+            ->setFrom('notifications@myth.forum', config('App')->siteName)
```

Comment:
The email address should be a config setting also, right?

Reply:
Yes, I forgot about that. I think we can use `config('Email')->fromEmail`, or just skip `setFrom()` altogether and fill in the `$fromEmail` and `$fromName` config variables - then the Email library should use those variables by default.